storage/redis/adapter.go

```diff
@@ -0,0 +1,101 @@
+package redis
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	r "gopkg.in/redis.v3"
+)
+
+type adapter struct {
+	bufferSize  int
+	redisClient *r.Client
+}
+
+// NewStorageAdapter returns a pointer to a new instance of a redis-based storage.Adapter.
+func NewStorageAdapter(bufferSize int) (*adapter, error) {
+	if bufferSize <= 0 {
+		return nil, fmt.Errorf("Invalid buffer size: %d", bufferSize)
+	}
+	db, err := getIntOpt("DEIS_LOGGER_REDIS_DB", 0)
```

Comment:
What you say is true. However, the various implementations of the `storage.Adapter` interface are instantiated through a factory. If this information were to be passed, then the factory (or worse, any client _of_ the factory) would have to parse these config values instead-- and that seems wrong. I'm trying to keep parsing of implementation-specific configuration close to where it's utilized.